src/lib/multiaddres.ts

```diff
@@ -27,89 +27,234 @@ import { multiaddr } from "@multiformats/multiaddr";
+  return multiaddrs.map((multiaddr) => {
+    return {
+      peerId: getPeerId(multiaddr),
+      multiaddr,
+    };
+  });
+}
+
+export const local: Node[] = multiaddrsToNodes(localMultiaddrs);
+
+const ADDR_MAP: Record<ContractsENV, Array<Node>> = {
   kras: krasnodar,
   stage,
   testnet: testNet,
+  local,
 };
 
-const resolveAddrs = (
-  maybeRelays: Relays | null | undefined,
-): Array<string> => {
-  if (maybeRelays === undefined || maybeRelays === null) {
-    return ADDR_MAP.kras.map((node) => {
-      return node.multiaddr;
-    });
+export async function ensureCustomRelays(fluenceConfig: FluenceConfig) {
+  const contractsEnv = await list({
+    message: "Select contracts environment for your custom network",
+    options: [...CONTRACTS_ENV],
+    oneChoiceMessage: (): never => {
+      throw new Error("Unreachable: only one contracts env");
+    },
+    onNoChoices: (): never => {
+      throw new Error("Unreachable: no contracts envs");
+    },
+  });
+
+  const fluenceEnvOrCustomRelays = commaSepStrToArr(
+    await input({
+      message: "Enter comma-separated list of relays",
+      validate: (input: string) => {
+        const relays = commaSepStrToArr(input);
+
+        if (relays.length === 0) {
+          return "You must specify at least one relay";
+        }
+
+        return true;
+      },
+    }),
+  );
+
+  fluenceConfig.customFluenceEnv = {
+    contractsEnv,
+    relays: fluenceEnvOrCustomRelays,
+  };
+
+  await fluenceConfig.$commit();
+  return fluenceEnvOrCustomRelays;
+}
+
+function getCustomRelays(
+  maybeFluenceConfigReadonly: FluenceConfigReadonly | null,
+) {
+  if (maybeFluenceConfigReadonly === null) {
+    commandObj.error(
+      `You must init fluence project if you want to use ${color.yellow(
+        "custom",
+      )} fluence env`,
+    );
+  }
+
+  if (maybeFluenceConfigReadonly.customFluenceEnv === undefined) {
+    commandObj.error(
+      `You must specify custom fluence env in ${color.yellow(
+        maybeFluenceConfigReadonly.$getPath(),
+      )} if you want to use ${color.yellow("custom")} fluence environment`,
+    );
   }
 
-  const relays = maybeRelays;
+  return maybeFluenceConfigReadonly.customFluenceEnv.relays;
+}
 
-  if (Array.isArray(relays)) {
-    return relays.map((relay) => {
-      return resolveRelay(relay);
-    });
+export function resolveRelays(
+  fluenceEnv: FluenceEnv,
+  maybeFluenceConfigReadonly: FluenceConfigReadonly | null,
+): Array<string> {
+  if (fluenceEnv === "custom") {
+    return getCustomRelays(maybeFluenceConfigReadonly);
   }
 
-  return ADDR_MAP[relays].map((node) => {
+  return ADDR_MAP[fluenceEnv].map((node) => {
     return node.multiaddr;
   });
-};
+}
 
-const getRandomArrayItem = <T>(ar: Array<T>): T => {
+function getRandomArrayItem<T>(ar: Array<T>): T {
   const randomIndex = Math.round(Math.random() * (ar.length - 1));
   const randomItem = ar[randomIndex];
   assert(randomItem !== undefined);
   return randomItem;
-};
+}
```

Comment:
I think this util function is too smart and specific.
Probably you can have a function for getting random integer from range and use that in your code.

Reply:
don't really understand what you are suggesting here. This function seems pretty small and to-the-point - it just takes a random item from array, don't really know how to decompose it further, if that's what you want

Reply:
Yes, the initial comment was about some decomposition. Add an additional function for getting random integer in rand and rewrite it in your source code.

```
const r = randInt(0, array.lenght);
const el = arr[r] || null;
```

But can we use instead of this utility function just `lodash.sample` ?

Reply:
yes. Just using lodash seems to be fine, I think I didn't have lodash as a dependency in the project when I wrote this function